src/app/options/options.component.spec.ts

```diff
@@ -93,18 +154,112 @@ describe('OptionsComponent', () => {
     expect((await options[0].getText())).toEqual("test1");
   });
 
+  it('should display mat-errors if input form control is blank', async () => {
+    await createState();
+    const inputError = fixture.debugElement.query(By.css('.input-error'));
+    const outputError = fixture.debugElement.query(By.css('.output-error'));
+    expect(component.inputControl.invalid).toEqual(true);
+    expect(component.outputControl.invalid).toEqual(true);
+    expect(inputError).toBeTruthy();
+    expect(outputError).toBeTruthy();
+  });
+
+  it('should display input mat-error if input form control is invalid', async () => {
+    await createState(ComponentState.INVALID_INPUT);
+    const inputError = fixture.debugElement.query(By.css('.input-error'));
+    expect(component.inputControl.invalid).toEqual(true);
+    expect(inputError).toBeTruthy();
+  });
+
+  it('should display output mat-error if output form control is invalid', async () => {
+    await createState(ComponentState.INVALID_OUTPUT);
+    const outputError = fixture.debugElement.query(By.css('.output-error'));
+    expect(component.outputControl.invalid).toEqual(true);
+    expect(outputError).toBeTruthy();
+  });
+
+  it('should not display input mat-error if input form control is valid', async () => {
+    await createState(ComponentState.VALID_INPUT);
+    const inputError = fixture.debugElement.query(By.css('.input-error'));
+    expect(component.inputControl.invalid).toEqual(false);
+    expect(inputError).toBeNull();
+  });
+
+  it('should not display output mat-error if output form control is valid', async () => {
+    await createState(ComponentState.VALID_OUTPUT);
+    const outputError = fixture.debugElement.query(By.css('.output-error'));
+    expect(component.outputControl.invalid).toEqual(false);
+    expect(outputError).toBeNull();
+  });
+
+  it('should update data$ observable from input in outputFormat box', async () => {
+    await createState();
+    const output = await loader.getHarness(MatAutocompleteHarness.with({
+      selector: '.outputInput'}));
+
+    await output.focus();
+    await output.enterText("Output format");
+
+    component.service.data$.subscribe(
```

Comment:
use testbed to inject the service as noted above. Make change here and everywhere else this applies.